lib/travis/build/script/shared/rvm.rb

```diff
@@ -15,30 +15,46 @@ module RVM
           rvm_remote_server_verify_downloads3=1
         )
 
+        DEFALUT_VERSION = 'default'
+
         def export
           super
-          sh.export 'TRAVIS_RUBY_VERSION', config[:rvm], echo: false if rvm?
+          # for now, we take the rvm key if both are defined
+          # see also:`#version` and `#cache_name_component`
+          if rvm?
+            sh.export 'TRAVIS_RUBY_VERSION', config[:rvm], echo: false
+          elsif config[:ruby]
+            sh.export 'TRAVIS_RUBY_VERSION', config[:ruby], echo: false
+          else
+            sh.export 'TRAVIS_RUBY_VERSION', DEFALUT_VERSION, echo: false
+          end
         end
 
         def setup
           super
-          setup_rvm if rvm?
+          setup_rvm unless chruby?
         end
 
         def announce
           super
           sh.cmd 'ruby --version'
-          sh.cmd 'rvm --version' if rvm?
+          sh.cmd 'rvm --version' unless chruby?
         end
 
         def cache_slug
-          super.tap { |slug| slug << "--rvm-" << ruby_version.to_s if rvm? }
+          super.tap { |slug| slug << "--#{cache_name_component}-" << ruby_version.to_s }
         end
 
         private
 
           def version
-            config[:rvm].to_s
+            if rvm?
+              config[:rvm].to_s
+            elsif chruby?
+              config[:ruby].to_s
+            else
+              DEFALUT_VERSION
```

Comment:
Looks like the typo is repeated in the rest of the file, so shouldn't cause errors.